### PR TITLE
🎉 Add amplicon architect

### PR DESCRIPTION
**Pull request by migbro**
## Description

Created tools needed for ecDNA analysis, Amplicon Architect.
Also created tools in preparation for a "grandfather" run
You can refer to the readme created for details on each tool, but the general workflow will end up being:
bam2cram(tools exists, no change needed)->cnvkit->cns2bed->prepare_aa->run_aa

Fixes # https://github.com/d3b-center/bixu-tracker/issues/1444

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] [cnvkit-batch](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/075764cf-5976-4391-94da-95e457d83c57/#)
- [ ] [cns-to-aa-bed](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/af4a9674-abc2-455b-9e5b-3884eb2a290d/)
- [ ] [prepare-aa - BAM IN](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/11576833-b256-4c83-8390-c0275b41a51c/)
- [ ] [prepare-aa - CRAM IN](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/591fb6f6-1a65-4759-85e6-72001db5c7db/_
- [ ] [amplicon-architect BAM IN](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/66d5a05b-f4c2-4c77-b9a6-655aa87b71cb/)
- [ ] [amplicon-architect CRAM IN](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/8dbdfbe0-cecf-41be-bf10-8f0a9d169cd0/)

**Test Configuration**:
* Environment: cavatica, cwltool (empty inputs test)
* Test files: A WGS bam and `.cnn` file from a previous CNVkit run

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have committed any related changes to the PR

### CODE REVIEW

File: docs/kfdrc_amplicon_architect.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Kids First DRC Amplicon Architect(AA) Workflow (WIP)
+This is an extra-chromosomal DNA (ecDNA) pipeline designed to detect ecDNA candidates.
+Briefly, chromosomal fragments in tumor cells make break off to form circular DNA.
```
Comment (sickler-alex) [inline docs/kfdrc_amplicon_architect.md:3]: I think "make" isn't needed here.

**Review by sickler-alex — APPROVED** [review]

File: tools/amplicon_architect.cwl [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        }
+      }
+  - position: 3
+    shellQuote: false
+    valueFrom: >-
+      && /home/programs/AmpliconArchitect-master/src/AmpliconArchitect.py
+inputs:
+  data_repo: { type: File, doc: "Reference tar ball obtained from https://datasets.genepattern.org/?prefix=data/module_support_files/AmpliconArchitect/", inputBinding: { position: 1} }
+  data_ref_version: { type: ['null', {type: enum, name: wgs_mode, symbols: ["GRCh38", "hg19", "GRCh37", "mm10", "GRCm38"]}], doc: "Genome version in data repo to use", default: "GRCh38", inputBinding: { position: 3, prefix: "--ref"} }
+  bam: { type: File, doc: "tumor bam file", secondaryFiles: ['^.bai?', '.crai?'], inputBinding: { position: 3, prefix: "--bam" } }
```
Comment (dmiller15) [inline tools/amplicon_architect.cwl:44]: These secondary file formats are not 1.2 compliant. You should use the pattern/required dict format.

File: tools/cnvkit_batch_only.cwl [hunk 3]
```diff
@@ -0,0 +1,63 @@
+cwlVersion: v1.0
```
Comment (dmiller15) [inline tools/cnvkit_batch_only.cwl:1]: Is this just a copy and that's why you're keeping the 1.0 version?

**Review by dmiller15 — APPROVED** [review]
Passes CWL tool and runs look good. Just noted a couple of CWL 1.0 vs 1.2 things.

File: tools/cnvkit_batch_only.cwl (hunk 3 above)
Comment (migbro) [reply tools/cnvkit_batch_only.cwl:1]: Yeah, it was a copy, I can make it 1.2 for the benefit of mankind

**Pull request merged**